pages/accountLists/[accountListId]/reports/financialAccounts/[financialAccountId]/Wrapper.tsx

```diff
@@ -32,28 +33,20 @@ export const FinancialAccountsWrapper: React.FC<Props> = ({ children }) => {
   const [activeFilters, setActiveFilters] =
     useState<FinancialAccountTransactionFilters>(urlFilters ?? {});
 
-  const [page, setPage] = useState<FinancialAccountPageEnum>(
-    FinancialAccountPageEnum.FinancialAccountPage,
+  const [financialAccountId, setFinancialAccountId] = useState(
+    getQueryParam(query, 'financialAccountId') ?? '',
   );
-  const [financialAccountId, setFinancialAccountId] = useState<
-    string | undefined
-  >(undefined);
 
-  const { financialAccount, searchTerm, accountListId } = query;
+  const { searchTerm } = query;
 
   useEffect(() => {
-    if (!financialAccount) {
-      setPage(FinancialAccountPageEnum.FinancialAccountPage);
+    if (!query.financialAccountId) {
       return;
     }
-    const length = financialAccount.length;
-    setFinancialAccountId(financialAccount[0]);
-    if (length === 1) {
-      setPage(FinancialAccountPageEnum.AccountSummaryPage);
-    } else if (length > 1) {
-      setPage(FinancialAccountPageEnum.AccountTransactionsPage);
+    if (typeof query.financialAccountId === 'string') {
+      setFinancialAccountId(query.financialAccountId);
     }
```

Comment:
You could use `getQueryParam` here to match the `useState`.

Reply:
I can use the `getQueryParam` but this causes a typescript error as it doesn't understand `query.financialAccountId` has a defined value, so I'm thinking of keeping it as `typeof query.financialAccountId === 'string'`

Reply:
That's fine